From ecd4d89ced2b42bd6f73226594cfdfd3185f8410 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:34:17 -0400
Subject: [alsa] hda/realtek: Correction of fixup codes for PB V7900 laptop

Message-id: <140972823573-323-patchset-jkysela@redhat.com>
Patchwork-id: 91019
O-Subject: [RHEL 7.1 PATCH 322/442] ALSA: hda/realtek - Correction of fixup codes for PB V7900 laptop
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Correcion of wrong fixup entries add in commit ca8f0424 to replace
static model quirk for PB V7900 laptop (will model).

[note: the removal of ALC260_FIXUP_HP_PIN_0F chain is also needed as a
part of the fix; otherwise the pin is set up wrongly as a headphone,
and user-space (PulseAudio) may be wrongly trying to detect the jack
state -- tiwai]

Fixes: ca8f04247eaa ('ALSA: hda/realtek - Add the fixup codes for ALC260 model=will')
Signed-off-by: Ronan Marquet <ronan.marquet@orange.fr>
Cc: <stable@vger.kernel.org> [v3.4+]
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: e30cf2d2bed3aed74a651c64de323ba26e4ff7d0)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index a00cd69..754fea2 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -1649,12 +1649,10 @@ static const struct hda_fixup alc260_fixups[] = {
  [ALC260_FIXUP_COEF] = {
   .type = HDA_FIXUP_VERBS,
   .v.verbs = (const struct hda_verb[]) {
-   { 0x20, AC_VERB_SET_COEF_INDEX, 0x07 },
-   { 0x20, AC_VERB_SET_PROC_COEF,  0x3040 },
+   { 0x1a, AC_VERB_SET_COEF_INDEX, 0x07 },
+   { 0x1a, AC_VERB_SET_PROC_COEF,  0x3040 },
    { }
   },
-  .chained = true,
-  .chain_id = ALC260_FIXUP_HP_PIN_0F,
  },
  [ALC260_FIXUP_GPIO1] = {
   .type = HDA_FIXUP_VERBS,
-- 
1.7.1